doc/telescope.txt

```diff
@@ -47,46 +47,38 @@ telescope.setup({opts})                                    *telescope.setup()*
-    history_handler: ~
-        This is a developer setting and is meant for an extension.
-
-        history_handler is a function that defines the behavior of the history.
-        Telescope core ships a rather simple implementation of it, with a unified
-        history across all pickers. Its not sensitive for cwd or picker.
-
-        We also offer an extension (nvim-telescope/telescope-smart-history.nvim)
-        which uses a sqlite3 database to store prompt into with the corresponding
-        metadata, like picker and cwd. This means cycling to previous entry will
-        return an entry that was actually used in this cwd. This is helpful if you
-        work on multiple projects at the same time.
-
-        It can be disabled the same way, by setting history_location to nil.
-
-        Expects function
-        Default: require('telescope.actions.history').get_simple_history
-
-                                         *telescope.defaults.history_limit*
-    history_limit: ~
-        You can set a limit on how many history elements will be stored.
-        Set to nil to disable it otherwise set to a number.
-
-        Default: nil
-
-                                      *telescope.defaults.history_location*
-    history_location: ~
-        This setting will enable result history. You can scroll though your
-        previous inputs. The mappings to do so are currently not mapped.
-        You can add them like this:
-
-          mappings = {
-            i = {
-              ["<C-Down>"] = actions.cycle_history_next,
-              ["<C-Up>"] = actions.cycle_history_prev,
+                                               *telescope.defaults.history*
+    history: ~
+          This field handles the configuration for prompt history. It can either
+          be a table or false.
+          If its false then history is disabled.
+
+          Currently mappings still need to be added, Example:
+            mappings = {
+              i = {
+                ["<C-Down>"] = require('telescope.actions').cycle_history_next,
+                ["<C-Up>"] = require('telescope.actions').cycle_history_prev,
+              },
             },
-          },
 
-        Default: nil
+          Fields:
+            - path:    the path to the telescope history as string
+                       default: stdpath("data")/telescope_history
+            - limit:   the amount of entries that will be written in the history
+                       if limit is set to nil it will grown unbound
+                       default: 100
+            - handler: is a developer setting that is meant for extensions.
+                       You can replace the actual implementation of the history
+                       and provide your own. One example of this is
+                       nvim-telescope/telescope-smart-history.nvim which allows
+                       context sensitive (cwd + picker) history.
```

Comment:
"handler": a Lua function that implements the history.
This is meant as a developer setting for extensions to override the history handling,
e.g., https://github.com/nvim-telescope/telescope-smart-history.nvim, which allows
context sensitive (cwd + picker) history.